### PR TITLE
[linker][new integer types] Fix missing case in dependency verifier

### PR DESCRIPTION
**Pull request by tnowacki**
- Fix missing case in dependency verifier for equality on signature tokens (from different contexts)
- Fix prop test case

## Motivation

- fixing bug in dependency verifier

## Test Plan

- new test

### CODE REVIEW

**Comment by oxade** [comment]
@tnowacki thanks for catching this.
There's just so many places that these types touch and I was wondering is there a proper way to do an audit to make sure no more missing enums?

File: language/move-bytecode-verifier/src/dependencies.rs [hunk 1]
```diff
@@ -472,7 +475,21 @@ fn compare_types(
                 Ok(())
             }
         }
-        _ => Err(PartialVMError::new(StatusCode::TYPE_MISMATCH)),
+        (SignatureToken::Bool, _)
+        | (SignatureToken::U8, _)
+        | (SignatureToken::U64, _)
```
Comment (oxade) [inline language/move-bytecode-verifier/src/dependencies.rs:480]: Thanks for expanding these out so future changes must be explicitly handled.

**Review by oxade — APPROVED** [review]
Thanks!

File: language/move-bytecode-verifier/transactional-tests/tests/dependencies/call_integers_valid.move [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//# publish
+module 0x42::m {
+public fun plus8(a: u8): u8 { a + 1 }
+public fun plus16(a: u16): u16 { a + 1 }
+public fun plus32(a: u32): u32 { a + 1 }
+public fun plus64(a: u64): u64 { a + 1 }
+public fun plus128(a: u128): u128 { a + 1 }
+public fun plus256(a: u256): u256 { a + 1 }
```
Comment (jiangying000) [inline language/move-bytecode-verifier/transactional-tests/tests/dependencies/call_integers_valid.move:8]: formatting issue here

Comment (tnowacki) [reply language/move-bytecode-verifier/transactional-tests/tests/dependencies/call_integers_valid.move:8]: It's just a test :) This is setup this way to make it easy to see each case

While the lack of newline after `{` is a bit obtuse. I've come to like not indenting inside of the `module` block. Not really a point in doing so IMO

**Comment by tnowacki** [comment]
> @tnowacki thanks for catching this. There's just so many places that these types touch and I was wondering is there a proper way to do an audit to make sure no more missing enums?

I don't think there is, sadly. Just writing tests :) This one is a bit subtle and not surprising we forgot it in the first PR. 

I did another pass, hence the fix for proptests.

**Review by sblackshear — APPROVED** [review]

**Pull request merged**